### ISSUE TITLE
[Bug]: UI not showing up on Gatekeeper(BepInEx 6.0.0-be.697+)

### ISSUE THREAD

**Issue by Robocraft999** [yukieiji/UnityExplorer#26]
### Are you on the latest version of UnityExplorer?

- [X] Yes, I'm on the latest version of UnityExplorer.

### Will other mods work? (Not mods that change settings, but small mods that access game objects, etc.)

- [ ] Yes, work fine!!

### Which release are you using?

BepInEx Unity IL2CPP CoreCLR

### Which game did this occur on?

Gatekeeper (Unity 2021.3.18f1)
The game is not modded by anybody else. As I still have to figure out to write an api for it. But I coded a simple mod which could include game code and was loaded successfully. I just don't know yet how to use it.

### Describe the issue.

I assume the window of UnityExplorer should appear on its on on startup. Nevertheless even upon pressing the toggle button (F7) nothing is howing up. I also changed the Keycode in the config with no success. I tried on BepInEx 6.0.0-be.697 IL2Cpp and BepInEx 6.0.0-be.725 IL2Cpp.

### Relevant log output

```shell
LogOutput.log: 
[Message: Preloader] BepInEx 6.0.0-be.725 - Gatekeeper (13.05.2024 17:15:10)
[Message: Preloader] Built from commit e1974e26fd7702c66b54c0d6879c90b988cc4920
[Info   :   BepInEx] System platform: Windows 10 64-bit
[Info   :   BepInEx] Process bitness: 64-bit (x64)
[Info   :   BepInEx] Running under Unity 2021.3.18f1
[Info   :   BepInEx] Runtime version: 6.0.7
[Info   :   BepInEx] Runtime information: .NET 6.0.7
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Warning:Il2CppInterop] Class::Init signatures have been exhausted, using a substitute!
[Info   :Il2CppInterop] Registered mono type Il2CppInterop.Runtime.DelegateSupport+Il2CppToMonoDelegateReference in il2cpp domain
[Info   :   BepInEx] 2 plugins to load
[Info   :   BepInEx] Loading [UnityExplorer 4.12.2]
[Message:UnityExplorer] UnityExplorer 4.12.2 initializing...
[Message:UnityExplorer] [UniverseLib] UniverseLib 1.5.7 initializing...
[Info   :Il2CppInterop] Registered mono type UniverseLib.UniversalBehaviour in il2cpp domain
[Info   :Il2CppInterop] Registered mono type UniverseLib.Runtime.Il2Cpp.Il2CppManagedEnumerator in il2cpp domain
[Message:UnityExplorer] [UniverseLib] Loaded Unhollowed modules in 0,002 seconds.
[Message:UnityExplorer] [UniverseLib] Setup deobfuscation cache in 0,1 seconds.
[Message:UnityExplorer] [UniverseLib] Finished UniverseLib initial setup.
[Info   :Il2CppInterop] Registered mono type UnityExplorer.ExplorerBehaviour in il2cpp domain
[Message:UnityExplorer] Initialized UnityCrashPrevention for: Canvas.get_renderingDisplaySize
[Info   :   BepInEx] Loading [My first plugin 1.0.0]
[Info   :My first plugin] Plugin Test2 is loaded!
[Info   :My first plugin] Things: 
[Info   :My first plugin] Plugin Test2 is loaded!
[Message:   BepInEx] Chainloader startup complete

Unity log (player.log) contains no errors and nothing like UnityExplorer or UniverseLib appears
```

**Comment by yukieiji** [comment]
Are you seeing any errors in the BepInEx.log?

**Comment by yukieiji** [comment]
I'am able to confirm the work with BepInEx 6.0.0.be725 in AmongUs.
So it seems to be a issue with the game.

**Comment by Robocraft999** [comment]
> Are you seeing any errors in the BepInEx.log?

where do I find this?

**Comment by yukieiji** [comment]
BepInEx/LogOutput.log

**Comment by Robocraft999** [comment]
Ah this is already appended see above

**Comment by yukieiji** [comment]
I can't find any logs regarding completion of initialization of UniverseLib etc. or creation of UI, are they all attached?

**Comment by Robocraft999** [comment]
I mean they are attached yes. I can repost them here but I don't think this helps:

```
LogOutput.log: 
[Message: Preloader] BepInEx 6.0.0-be.725 - Gatekeeper (13.05.2024 17:15:10)
[Message: Preloader] Built from commit e1974e26fd7702c66b54c0d6879c90b988cc4920
[Info   :   BepInEx] System platform: Windows 10 64-bit
[Info   :   BepInEx] Process bitness: 64-bit (x64)
[Info   :   BepInEx] Running under Unity 2021.3.18f1
[Info   :   BepInEx] Runtime version: 6.0.7
[Info   :   BepInEx] Runtime information: .NET 6.0.7
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Warning:Il2CppInterop] Class::Init signatures have been exhausted, using a substitute!
[Info   :Il2CppInterop] Registered mono type Il2CppInterop.Runtime.DelegateSupport+Il2CppToMonoDelegateReference in il2cpp domain
[Info   :   BepInEx] 2 plugins to load
[Info   :   BepInEx] Loading [UnityExplorer 4.12.2]
[Message:UnityExplorer] UnityExplorer 4.12.2 initializing...
[Message:UnityExplorer] [UniverseLib] UniverseLib 1.5.7 initializing...
[Info   :Il2CppInterop] Registered mono type UniverseLib.UniversalBehaviour in il2cpp domain
[Info   :Il2CppInterop] Registered mono type UniverseLib.Runtime.Il2Cpp.Il2CppManagedEnumerator in il2cpp domain
[Message:UnityExplorer] [UniverseLib] Loaded Unhollowed modules in 0,002 seconds.
[Message:UnityExplorer] [UniverseLib] Setup deobfuscation cache in 0,1 seconds.
[Message:UnityExplorer] [UniverseLib] Finished UniverseLib initial setup.
[Info   :Il2CppInterop] Registered mono type UnityExplorer.ExplorerBehaviour in il2cpp domain
[Message:UnityExplorer] Initialized UnityCrashPrevention for: Canvas.get_renderingDisplaySize
[Info   :   BepInEx] Loading [My first plugin 1.0.0]
[Info   :My first plugin] Plugin Test2 is loaded!
[Info   :My first plugin] Things: 
[Info   :My first plugin] Plugin Test2 is loaded!
[Message:   BepInEx] Chainloader startup complete

Unity log (player.log) contains no errors and nothing like UnityExplorer or UniverseLib appears
```

**Comment by yukieiji** [comment]
When UE is started in AmongUs, the log is as follows
You will see that the log shows that the initialization is complete as indicated by` [Message:UnityExplorer] UnityExplorer 4.12.1 (IL2CPP) initialized.`.
The fact that no such indication is shown indicates that some kind of initialization has failed.

```
[Message: Preloader] BepInEx 6.0.0-be.725 - Among Us (2023/10/25 2:20:03)
[Message: Preloader] Built from commit e1974e26fd7702c66b54c0d6879c90b988cc4920
[Info   :   BepInEx] System platform: Windows 10 64-bit
[Info   :   BepInEx] Process bitness: 32-bit (x86)
[Info   :   BepInEx] Running under Unity 2020.3.45f1
[Info   :   BepInEx] Runtime version: 6.0.7
[Info   :   BepInEx] Runtime information: .NET 6.0.7
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Info   :Il2CppInterop] Registered mono type Il2CppInterop.Runtime.DelegateSupport+Il2CppToMonoDelegateReference in il2cpp domain
[Info   :   BepInEx] 1 plugin to load
[Info   :   BepInEx] Loading [UnityExplorer 4.12.1]
[Message:UnityExplorer] UnityExplorer 4.12.1 initializing...
[Message:UnityExplorer] [UniverseLib] UniverseLib 1.5.6 initializing...
[Info   :Il2CppInterop] Registered mono type UniverseLib.UniversalBehaviour in il2cpp domain

[Info   :Il2CppInterop] Registered mono type UniverseLib.Runtime.Il2Cpp.Il2CppManagedEnumerator in il2cpp domain
[Message:UnityExplorer] [UniverseLib] Loaded Unhollowed modules in 0.003 seconds.
[Message:UnityExplorer] [UniverseLib] Setup deobfuscation cache in 0.27100003 seconds.
[Message:UnityExplorer] [UniverseLib] Finished UniverseLib initial setup.
[Info   :Il2CppInterop] Registered mono type UnityExplorer.ExplorerBehaviour in il2cpp domain
[Message:UnityExplorer] Initialized UnityCrashPrevention for: Canvas.get_renderingDisplaySize
[Message:   BepInEx] Chainloader startup complete
[Message:UnityExplorer] [UniverseLib] Initialized Legacy Input support
[Message:UnityExplorer] [UniverseLib] 	 IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.Cursor.lockState
[Message:UnityExplorer] [UniverseLib] 	 IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.AssetBundle.UnloadAllAssetBundles
[Info   :Il2CppInterop] Registered mono type UniverseLib.AssetBundle in il2cpp domain
[Message:UnityExplorer] [UniverseLib] Loaded modern bundle for Unity 2020.3.45f1
[Message:UnityExplorer] [UniverseLib] UniverseLib 1.5.6 initialized.
[Message:UnityExplorer] Creating UI...
[Message:UnityExplorer] UnityExplorer 4.12.1 (IL2CPP) initialized.
```

**Comment by yukieiji** [comment]
I made some changes in v4.12.2 regarding the initialization process around the input, so maybe it will work in v4.12.1.

**Comment by Robocraft999** [comment]
Unfortunatly it doesn't. Same output with same missing lines of output. I tried to run it with Among us as well but BepInEx doesn't even start there.

**Comment by yukieiji** [comment]
I'm not sure why it won't start with AmongUs, but is this game running anti-cheat or something? Or maybe that's what's preventing UE from launching.

**Comment by Robocraft999** [comment]
Among us works now. With UnityExplorer

**Comment by yukieiji** [comment]
I understand about the AmongUs.
this game running anti-cheat or something? Or maybe that's what's preventing UE from launching.

**Comment by Robocraft999** [comment]
nothing that I know about. It's a smaller Indie game. It's a bit like Risk of Rain. This kind of games usually don't have anti cheat and don't need one in my opinion

**Comment by yukieiji** [comment]
Hmm...,Tried the solution described [here](https://github.com/yukieiji/UnityExplorer?tab=readme-ov-file#common-issues-and-solutions) and it doesn't work?

**Comment by Robocraft999** [comment]
> Hmm...,Tried the solution described [here](https://github.com/yukieiji/UnityExplorer?tab=readme-ov-file#common-issues-and-solutions) and it doesn't work?

yep both doesn't work. I also noticed in the Among Us plugins folder there are some files and folders generated (Logs, Scripts and data.cfg) and in the Gatekeeper plugins folder is nothing except the two dlls. So where ever it fails it doesn't even get to the part of generating these

**Comment by yukieiji** [comment]
If the directory is not created, then [this process](https://github.com/yukieiji/UnityExplorer/blob/aa3030ab0d6b28a45f13539a9a25e7f689be3c35/src/ExplorerCore.cs#L45) has failed.
The fact that this process in C# is failing suggests that there is some kind of block process running.

**Comment by Robocraft999** [comment]
I also tried to use MelonLoader and found something out: When running UE with MelonLoader with debug enabled. There is this message: `[ML Core] Received Quit Request! Shutting down...`. I also noticed that when running Gatekeeper with BepInEx the console closes before the game is opening (even when setting `PreventClose = true`). Same with MelonLoader. When I was testing Among Us with BepInEx the console stayed open. Maybe the game forces a restart or something so that the process the modloader is running with gets shut down and a new one without is started. But I also have to say that I was not modifying the main installation folder but modifying them in seperate folders
